### PR TITLE
Math Magicians App: Refactor classes to React hooks.

### PR DESCRIPTION
**Pull request by smunozmo**
# Documentation

All the documentation is on the README file.

## Updates:

*  Refactored class based components to hooks.

## Notes:

✅  - Used  React.
✅  - No linters errors.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,62 +1,63 @@
...
+
+  return (
+    <div className="App">
+      <table>
+        <td>
+          <form type="text" value="0" className="display">0</form>
+          <br />
+          <button type="button" value="AC" onClick={UpdateObj}>AC</button>
+          <button type="button" value="+" onClick={UpdateObj}>+/-</button>
+          <button type="button" value="%" onClick={UpdateObj}>%</button>
```
Comment (DeliceLydia) [inline src/components/Calculator.js:40]: - [x] Currently `%` button is not working as it supposed please work on that

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,62 +1,63 @@
...
+  };
+
+  return (
+    <div className="App">
+      <table>
+        <td>
+          <form type="text" value="0" className="display">0</form>
+          <br />
+          <button type="button" value="AC" onClick={UpdateObj}>AC</button>
+          <button type="button" value="+" onClick={UpdateObj}>+/-</button>
```
Comment (DeliceLydia) [inline src/components/Calculator.js]: - [x] Please work on the functionality of `+/-` currently is not working

File: src/components/Calculator.js [hunk 3]
```diff
@@ -1,62 +1,63 @@
...
+    calcObjUpdate(currentObj);
+  };
+
+  return (
+    <div className="App">
+      <table>
+        <td>
+          <form type="text" value="0" className="display">0</form>
+          <br />
+          <button type="button" value="AC" onClick={UpdateObj}>AC</button>
```
Comment (DeliceLydia) [inline src/components/Calculator.js:38]: - [x] I recommend to display a default `zero` when you click on this `AC`

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Hi @smunozmo👏,

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 2 above)
Comment (DeliceLydia) [reply src/components/Calculator.js]: Well resolved this!

File: src/components/Calculator.js (hunk 3 above)
Comment (DeliceLydia) [reply src/components/Calculator.js:38]: Resolved it! Good job..

File: src/components/Calculator.js (hunk 1 above)
Comment (DeliceLydia) [reply src/components/Calculator.js:40]: - [x] Currently `%` button still showing `underfined` please fix that and do not give up

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Hi @smunozmo👏,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

> They say " Feedbacks are the breakfast of the champions" don't give up!


 ## Required Changes ♻️
_Check the comments I added under the previous review._




### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by smunozmo** [comment]
Hi @DeliceLydia 

Thank you for your comments, I tried these test cases for the Modulo JS math function (%) in order to calculate the remainder of an integer division (as the operate.js file uses one.mod(two).toString() for that operation).

I recorded a video with these test cases: 
https://www.loom.com/share/bf7acfc7eabb472aa8e034b4b63a2857

btn 8 -> btn % -> btn 2 -> btn = -> returns '2'
btn 9 -> btn % -> btn 3 -> btn = -> returns '0'
btn 10-> btn % -> btn 3 -> btn = -> returns '1'
btn 11 -> btn % -> btn 4 -> btn = -> returns '3'
btn 12 -> btn % -> btn 5 -> btn = -> returns '2'
btn 3 -> btn % -> btn 3 -> btn = -> returns '0'

As you can see in the video the  % button (.mod function) is working.

Kindly let me know what case is giving you an undefined result, I will try to reproduce the case but this is my last review attempt.

Happy coding! 🤓

**Review by DeliceLydia — APPROVED** [review]
# Status: Approved🎉 🎉 

## Hi @smunozmo, 

Your project is complete! and I commend you on how you communicate with a reviewer in a polite and professional way. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**